src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -1619,8 +1625,8 @@ internal void ClearTransaction(Exception? disposeReason = null)
         (sender, certificate, chain, sslPolicyErrors)
             => true;
 
-    static RemoteCertificateValidationCallback SslRootValidation(string certRootPath, bool verifyFull) =>
-        (sender, certificate, chain, sslPolicyErrors) =>
+    static RemoteCertificateValidationCallback SslRootValidation(string? certRootPath, bool verifyFull, X509Certificate2? caCertificate)
```

Comment:
nit: only because caCertificate and certRootPath belong together.

```suggestion
    static RemoteCertificateValidationCallback SslRootValidation(bool verifyFull, string? certRootPath, X509Certificate2? caCertificate)
```